runestone/common/js/renderComponent.js

```diff
@@ -18,45 +18,48 @@ export function renderRunestoneComponent(componentSrc, whereDiv, moreOpts) {
     let componentKind = $($(`#${whereDiv} [data-component]`)[0]).data(
         "component"
     );
-    let opt = {};
-    opt.orig = jQuery(`#${whereDiv} [data-component]`)[0];
-    if (opt.orig) {
-        opt.lang = $(opt.orig).data("lang");
-        opt.useRunestoneServices = true;
-        opt.graderactive = false;
-        opt.python3 = true;
-        if (typeof moreOpts !== "undefined") {
-            for (let key in moreOpts) {
-                opt[key] = moreOpts[key];
+    // Import the JavaScript for this component before proceeding.
+    runestone_import(componentKind).then(() => {
```

Comment:
Lazy Question: could this be done as an await for consistency with the our use of async/await  in the rest of the components?

Reply:
It's not an `async` function, so I thought it would be easiest to use a `.then()` instead.

Reply:
I mean the `export function renderRunestoneComponent(componentSrc, whereDiv, moreOpts) {` which contains this code.

Reply:
But this change gives it async behavior yes?

I like to declare these functions async to avoid future mistakes thinking that I can call the function and expect synchronous results.

Reply:
True -- the component will sometimes be rendered later instead of synchronously. Would you like me to update that?

Reply:
That would be great.

I'm not sure we are 100% compliant with that rule, but we are getting closer, and I can add that to the CONTRIBUTING file after this is merged so future contributors will keep that in mind.

Reply:
OK, done!